### PR TITLE
The correct instance type of the created Order

### PR DESCRIPTION
**Pull request by Seleda**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | "Self" in this case in the "OrderCore" class, and the instance type will be "OrderCore". The correct type must be "Order" to use in Prestashop.
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/20905)

### CODE REVIEW

**Comment by Progi1984** [comment]
@Seleda Thank you for your contribution. :hugs: 

Could you explain in [an issue your initial bug for this PR](https://github.com/PrestaShop/PrestaShop/issues/new/choose) (or may be link to an issue) ?
Could you give us a scenario which permits to test the bug before and after this PR ?

Thanks :+1:

File: classes/order/Order.php [hunk 1]
```diff
@@ -1173,7 +1173,7 @@ public static function getByCartId($id_cart)
     {
         $id_order = (int) self::getIdByCartId((int) $id_cart);
 
-        return ($id_order > 0) ? new self($id_order) : null;
+        return ($id_order > 0) ? new Order($id_order) : null;
```
Comment (PierreRambaud) [inline classes/order/Order.php]:
```suggestion
        return ($id_order > 0) ? new static($id_order) : null;
```

If you want to return the same instance type, use `static` instead

Comment (Seleda) [reply classes/order/Order.php]: For example: OrderSlip::create() requires an instance of "Order". And many other places.

Comment (Seleda) [reply classes/order/Order.php]: new static($id_order) well

**Review by atomiix — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
As QA is not needed here I merge it!

Thanks @Seleda

**Pull request merged**